pkg/context/context.go

```diff
@@ -111,12 +115,10 @@ func (ctx *Context) Setup() error {
 // You should always call `Teardown` if you called `Setup` to free any resources
 // acquired by `Setup`. Check `Do` for more automated management.
 func (ctx *Context) Teardown() error {
-	if ctx.SnapshotRoot != "" {
-		// if the client code provided the unpack directory,
-		// then it is also in charge of the cleanup.
+	if len(ctx.snapshotUnpackedPath) == 0 {
```

Comment:
mostly curious, why just not
```go
if ctx.snapshotUnpackedPath == "" {
```

Reply:
because of a personal preference, I don't like to use string literals. If golang would have had a strings.Empty constant and specify it is the default string value, it would be fine, otherwise it has a "smell" (using literlas in code).
The joke is on me since the integer 0 is also a literal.

You can also have a look on the 'tabs vs spaces' like (https://www.youtube.com/watch?v=SsoOG6ZeyUI) "check empty strings" google  chat:
   https://groups.google.com/g/golang-nuts/c/7Ks1iq2s7FA?pli=1
here is an interesting quote from Russ Cox extracted from the above discussion:
 `
The one that makes the code clear.
If I'm about to look at element x I typically write
len(s) > x, even for x == 0, but if I care about
"is it this specific string" I tend to write s == "".
It's reasonable to assume that a mature compiler will compile
len(s) == 0 and s == "" into the same, efficient code.
Right now 6g etc do compile s == "" into a function call
while len(s) == 0 is not, but that's been on my to-do list to fix.
`

Reply:
thanks, makes sense. I don't have a strong opinion here so I don't mind.

Reply:
@marcel-apf my rule of thumb is to generally use the style of code already present in the codebase :) If you wouldn't mind using the `== ""` style, I'd appreciate that.

Reply:
done